Dev

VincentHuet opened the pull request:
Ce que j'ai fais permet de : 
- référencer un chemin en lui donnant un nom et en indiquant son chemin
- charger le comtenu d'un fichier yml dans la db
- enregistrer des traducteurs via devise
- enregistrer des locales

Commit:
look + liste des clefs + liste des sources + listes des locales
manque fonctionnalité pour création de traduction

Commit:
reste à finaliser la traduction

Commit:
version opérationnelle reste quelques ajustements

tanguygo commented:
Pas besion d'avoir les gliphicons, ils sont d'offices inclus dans la gem normalement.

tanguygo commented on app/assets/stylesheets/style.css.less:
```diff
@@ -0,0 +1,53 @@
+  border-radius: @radius;
+  -webkit-border-radius: @radius;
+  -moz-border-radius: @radius;
+}
```
Un mixin utilise déjà dans Twitter Bootstrap.

.border-radius(5px);

tanguygo replied:
Faut inclures le fichier mixins pour pouvoir l'utiliser par contre, je t'invite à aller le lire sur leur repository.

@import "twitter/bootstrap/mixins";

tanguygo commented on app/controllers/translations_controller.rb:
```diff
@@ -25,7 +28,18 @@ def show
+
+
+
     @translation = Translation.new
```
Remove empty line.

tanguygo commented on app/controllers/translations_controller.rb:
```diff
@@ -25,7 +28,18 @@ def show
+    @translation.phrase_id = @corresponding_phrase.id
+
+    @translation.author = current_translator.name
+
```
Il ne faut pas l'initialiser dans le new car celui-ci peut être changer directement dans le input field. La variable ne peut donc pas être trustée.

Il faut juste le mettre dans le create.

De plus il faudrait lier une translation à un author_id plutôt qu'à une string.

tanguygo commented on app/controllers/translations_controller.rb:
```diff
@@ -25,7 +28,18 @@ def show
     @translation = Translation.new
+    
+    @translation.phrase_id = @corresponding_phrase.id
+
```
Si une phrase has_many translations
@translation = @coressponding_phrase.phrases.build()

tanguygo commented on app/controllers/translations_controller.rb:
```diff
@@ -41,8 +55,17 @@ def edit
+
+    @translation.locale_id = current_translator.locale_id
+
+
```
Même chose qu'au dessus.

tanguygo commented on app/helpers/yml_sources_helper.rb:
```diff
@@ -36,17 +36,17 @@ def parse_yaml(h, hash_path, indent)
 
-        new_phrase.yaml_path = hash_path + "/" 
+        new_phrase.yaml_path = hash_path + "/" + key2.to_s
         new_phrase.key = key2.to_s
```
Remplacer par un ".".

tanguygo commented on app/models/translation.rb:
```diff
@@ -2,20 +2,17 @@
 class Translation < ActiveRecord::Base
-  attr_accessible :Primary_updated, :locale, :locale_id, :phrase, :phrase_id, :previous_text, :text
+  attr_accessible :Primary_updated, :locale_id, :phrase_id, :previous_text, :text
 
```
Virer la majuscule.

tanguygo commented on app/views/translations/show.html.haml:
```diff
@@ -1,25 +1,17 @@
-  = @translation.locale_id
+  %b Original text:
+  = Phrase.find(@translation.phrase_id).en_value
 %p
```
Bizarre cette méthode en_value.

tanguygo commented on db/schema.rb:
```diff
@@ -29,20 +29,39 @@
     t.datetime "updated_at", :null => false
     t.string   "yaml_path"
+    t.string   "en_value"
   end
```
Comprend pas l'intérêt du en_value.

Ca doit aller dans une translation cette en_value.

Commit:
modification complète suite aux demandes + manques modification suite aux retours de Tanguy

Commit:
Fin des ajustements fonctionnement selon consigne
Passage sous rails best pratice

Commit:
nested_routes:ok
countdown_modification:ok
needed_translation_flag:nok
mailer:nok
html_uploader:nok
Yml_generator:nok
